### PR TITLE
WEB-1691: Switch to @khanacademy/eslint-config for our shared linting config

### PR DESCRIPTION
**Pull request by kevinbarabash**
This PR also updates .eslintrc to .eslintrc.js so that we can have proper comments in our custom config for this repo.

Test Plan: test rules from each of the four .js files.

**react.js**
Screen Shot 2019-12-06 at 5 52 47 PM

**index.js**
Screen Shot 2019-12-06 at 5 51 29 PM

**flow.js**
Screen Shot 2019-12-06 at 5 50 57 PM

**shared.js**
Screen Shot 2019-12-06 at 6 13 49 PM

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 71bc1f51693e772670fb767473f1d90fd26c339a

https://deploy-preview-629--wonder-blocks.netlify.com

**Comment by somewhatabstract** [comment]
> The code changes themselves seem reasonable to me

Me too! 

> I think we need a more robust test plan

Agreed.

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = require("@khanacademy/eslint-config/.prettierrc.js");
```
Comment (kevinbarabash) [inline .prettierrc.js]: @jeresig I made the following change after you approved.  This is to ensure that we're using the same prettier config for VSCode as we are for eslint.

Comment (jeresig) [reply .prettierrc.js]: Oh nice!

**Pull request merged**